app/services/membership/withdraw_service.rb

```diff
@@ -15,6 +16,10 @@ def self.call(membership)
 
   def perform
     raise MembershipMustHaveDefaultPost unless membership.has_post?(PostType::DEFAULT_POST_ID)
+    raise MembershipHasPointRequests unless PointRequest.where(
+                                                          user: membership.user,
+                                                          evaluation: membership.group.evaluations)
+                                                        .count.zero?
```

Comment:
You can avoid weird indentation and help understanding by moving this to a method. What do you think?

```suggestion
    raise MembershipMustHaveDefaultPost unless membership.has_post?(PostType::DEFAULT_POST_ID)
    raise MembershipHasPointRequests if point_request_exists?
```

```ruby
def point_request_exists?
  PointRequest.exists?(user_id: membership.user_id, evaluation: membership.group.evaluations)
end
```

There is also a little change in the query, by referencing `user_id` instead of `user.id`, a db lookup can be avoided. I haven't checked though if the user is already loaded and cached at this point.